Added sha_hash function which returns sha for the given algorithm.

shantanoo opened the pull request:
Instead of `sha256_hash` function, added new function `sha_hash`. Pass the sha function name (algorithm) from hashlib along with the string to get the hash. Supported algorithms are sha1, sha224, sha256, sha384, sha3_224, sha3_256, sha3_384, sha3_512, sha512.

Branches to be merged in:
develop
version-15

blaggacao committed:
feat: add debug tests

cogk committed:
fix(File): Unquote file_url only once

cogk committed:
fix(File): Disallow percent sign in file_name

cogk committed:
fix(InboundEmail): Unquote percent-encoded file names

cogk committed:
test(File): Add test for attachments with percent-encoded file names

blaggacao committed:
feat: subject processing to global server script config

blaggacao committed:
feat: add cache control

blaggacao committed:
feat: make more resources in the critical path cacheable

blaggacao committed:
feat(utm): make data helper more useful

blaggacao committed:
fix(notification): add ability to enqueue-call from server scripts

shantanoo committed:
Added sha_hash function which returns sha for the given algorithm. Supported alorithms are sha1, sha224, sha256, sha384, sha3_224, sha3_256, sha3_384, sha3_512, sha512

shantanoo committed:
Added sha_hash function which returns sha for the given algorithm. Supported alorithms are sha1, sha224, sha256, sha384, sha3_224, sha3_256, sha3_384, sha3_512, sha512

shantanoo committed:
Removed white-spaces

shantanoo committed:
Removed white-spaces

shantanoo committed:
Removed extra whitespace

shantanoo committed:
Removed extra whitespace

shantanoo committed:
Added tests for sha_hash

shantanoo committed:
Added tests for sha_hash

akhilnarang commented:
Any specific reason for wanting such a function? Its basically just a wrapper, not adding any functionality really.

shantanoo commented:
Need sha checksum function in server script for communicating with payment gateway. `sha256_hash` is already there, but not exposed. Created a wrapper for all available sha algorithms and exported them so that it is accessible in server script.

Ponnusamy1-V committed:
fix: time picker doesn't show correct value

akhilnarang committed:
Merge pull request #27528 from akhilnarang/escape-html-awesomebar

fix(search_utils): escape HTML before displaying in awesomebar

blaggacao committed:
Merge pull request #27237 from blaggacao/feat/datatable-tree-depth-on-query-reports

feat: set variable datatable tree depth

akhilnarang committed:
Merge pull request #27515 from Ninad1306/map_docs_fix

fix: Update Child table from the Last Source Doc

gavindsouza committed:
fix(website_search): Avoid recursive loop if _items_to_index is falsy

cogk committed:
fix: Set undefined values to zero in round_floats_in

akhilnarang committed:
fix(sessions): logout properly when user requests logout on password change

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27541 from cogk/fix-round_floats_in-ensure-values-are-set-to-zero

fix: Set undefined values to zero in round_floats_in

akhilnarang committed:
Merge pull request #27539 from frappe/gavindsouza/website_search-norerun

fix(website_search): Avoid recursive loop if _items_to_index is falsy

blaggacao committed:
Merge pull request #26752 from blaggacao/perf/make-get-requests-potentially-cachable

perf: make get requests cacheable

blaggacao committed:
chore: ignore helix folder

barredterra committed:
Merge branch 'develop' into address-autocomplete

gavindsouza committed:
fix!: Skip virtual fields in meta.get_valid_columns

git-avc committed:
Merge branch 'frappe:develop' into fix-report-excel-translation

gavindsouza committed:
fix: Use doc.get to safely check for attr

This bypasses the bungle during site creations when meta isn't present in
the database yet

akhilnarang committed:
Merge pull request #23691 from phot0n/fix-email-bulk-retry

refactor(minor): email retry

barredterra committed:
chore: bump pycountry to v24.6.1

gavindsouza committed:
fix: Separate meta.get_valid_fields from *columns

shantanoo commented:
@akhilnarang any suggestions? or can this be merged?

akhilnarang committed:
Merge pull request #27542 from akhilnarang/clear-all-user-sessions

fix(sessions): logout properly when user requests logout on password change

barredterra committed:
fix(query report)!: skip empty filters in print (#27016)

gavindsouza committed:
fix: Set doc title name as fallback

When the title field is unset for a document, it's value gets unset in a
link field on click. Added this fallback on the endpoint so that
clicking on field doesn't add an additional API call to fetch the title
via Desk.

Co-authored-by: Marica Dsouza <maricadsouza221197@gmail.com>

shariquerik committed:
fix: changed default check icon

blaggacao committed:
Merge pull request #27557 from blaggacao/chore/fix-editorconfig-json

chore: ignore helix folder

rohitwaghchaure committed:
fix: label for web form

shariquerik committed:
fix: consider default path for Website User if applicable

shariquerik committed:
fix: added apps data in boot

shariquerik committed:
fix: show Apps if multiple app or show CRM if only CRM is installed in dropdown

gavindsouza committed:
fix: Pass doc.name as label fallback in search_link

Populate desk link titles when relevant information is available via
search_link endpoint

shariquerik committed:
fix: redirect to /app if no apps are installed

shariquerik committed:
test: fixed failing user test

shariquerik committed:
Merge pull request #27571 from shariquerik/apps-switcher-fix-1

fix: Consider default app path for Website User if applicable

shariquerik committed:
fix: boot.default_path is moved to boot.apps_data.default_path

akhilnarang committed:
fix(assign_to): escape HTML

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

shariquerik committed:
Merge pull request #27577 from shariquerik/apps-page-fix-1

fix: boot.default_path is moved to boot.apps_data.default_path

akhilnarang committed:
fix: don't allow uninstalling frappe...

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27580 from akhilnarang/disallow-uninstalling-frappe

fix: don't allow uninstalling frappe...

deepeshgarg007 committed:
feat: Custom timeouts for prepared report

dependabot[bot] committed:
chore(deps): bump micromatch from 4.0.5 to 4.0.8

Bumps [micromatch](https://github.com/micromatch/micromatch) from 4.0.5 to 4.0.8.
- [Release notes](https://github.com/micromatch/micromatch/releases)
- [Changelog](https://github.com/micromatch/micromatch/blob/master/CHANGELOG.md)
- [Commits](https://github.com/micromatch/micromatch/compare/4.0.5...4.0.8)

---
updated-dependencies:
- dependency-name: micromatch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

maharshivpatel committed:
fix: on load of beta print format set iframe height

there was a edge case where when print format beta was set to default which never ran code which sets the size for container that holds format.
fixed that by make iframe height to 100vh minus navbar and heading on load of beta print format

akhilnarang committed:
Merge pull request #27586 from frappe/custom_report_timeout

feat: Custom timeouts for prepared report

akhilnarang committed:
Merge pull request #27597 from maharshivpatel/fix-beta-on-load

fix: on load of beta print format set iframe height

barredterra committed:
Merge remote-tracking branch 'upstream/develop' into user_doctype_permissions

akhilnarang committed:
Merge pull request #27570 from rohitwaghchaure/fixed-label-button

fix: label for web form

akhilnarang committed:
Merge pull request #27578 from akhilnarang/sanitize-html

fix(assign_to): escape description HTML

nikkothari22 committed:
fix: increase Google Calendar Event ID character limit

sumitbhanushali committed:
Merge pull request #27529 from frappe/fix_child_depends_on

fix: apply depends_on on field update

akhilnarang committed:
fix(sessions): pass `force` parameter to `get_sessions_to_clear()`

Not sure why this didn't get committed with the original PR: #27542

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27553 from gavindsouza/rm-virtual-df-from-get_valid_columns

fix!: Skip virtual fields in meta.get_valid_columns

akhilnarang committed:
Merge pull request #27614 from akhilnarang/fix-27542

fix(sessions): pass `force` parameter to `get_sessions_to_clear()`

pu-raihan committed:
fix: usability of Phone field (#27076)

barredterra committed:
refactor: import validation utils

barredterra committed:
fix: skip all validation if there's no data

marination committed:
fix: `get_valid_fields` excludes `show_on_timeline`, breaking migrations

Co-authored-by: Gavin <gavin.dsouza@switchup.de>

rmehta committed:
feat: App switcher

rmehta committed:
fix(minor): bring back copy to clipboard

frappe-pr-bot committed:
fix: sync translations from crowdin (#27471)

rmehta committed:
fix(minor): page creation

marination committed:
feat(minor): Allow HTML descriptions on Grids (#27620)

frappe-pr-bot committed:
chore: update POT file (#27590)

rmehta committed:
fix(minor): add website in app switcher

shariquerik committed:
fix: only change status on communication received if status is Replied & status contain option Open

shariquerik committed:
Merge pull request #27627 from shariquerik/communication-fix

fix: only change status on communication received if status is Replied & status contain option Open

cogk committed:
fix(email): Try to parse bad email address headers

balamurali27 committed:
fix(EmailQueue): Log more error onto email queue

Require it log actual exception raised by Email Delivery Service on plan
limit reached

akhilnarang committed:
Merge pull request #27632 from frappe/email-queue-more-error

fix(EmailQueue): Log more error onto email queue

akhilnarang committed:
Merge pull request #27619 from marination/meta-migrate

fix: `get_valid_fields` excludes `show_on_timeline`, breaking migrations

blaggacao committed:
feat: add unit tests for URL tracker functions

blaggacao committed:
feat(analytics): add basic utm nomenclature for analytics

blaggacao committed:
feat(analytics): add to website workspace

blaggacao committed:
chore: ignore aider files

blaggacao committed:
Merge pull request #27639 from blaggacao/chore/ignore-aider-files

chore: ignore aider files

blaggacao committed:
Merge pull request #26938 from blaggacao/fix/enqueue-notifications-from-scripts

fix(notification): add ability to enqueue-call from server scripts

dependabot[bot] committed:
chore(deps): bump actions/download-artifact in /.github/workflows

Bumps [actions/download-artifact](https://github.com/actions/download-artifact) from 3 to 4.1.7.
- [Release notes](https://github.com/actions/download-artifact/releases)
- [Commits](https://github.com/actions/download-artifact/compare/v3...v4.1.7)

---
updated-dependencies:
- dependency-name: actions/download-artifact
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

giulio-perosino committed:
fix: background for pagination buttons on dark theme

akhilnarang committed:
chore(deps): bump `cryptography`

(and `pyOpenSSL` to go along with it)

`pip-audit` is complaining about this
Resolves https://github.com/advisories/GHSA-h4gh-qq45-vh27

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27640 from frappe/dependabot/github_actions/dot-github/workflows/actions/download-artifact-4.1.7

chore(deps): bump actions/download-artifact from 3 to 4.1.7 in /.github/workflows

akhilnarang committed:
Merge pull request #27629 from cogk/fix-parse-worse-email-headers

fix(email): Try to parse bad email address headers

akhilnarang committed:
Merge pull request #27643 from giulio-perosino/develop

fix(style): background for pagination buttons on dark theme

akhilnarang committed:
Merge pull request #27589 from frappe/dependabot/npm_and_yarn/micromatch-4.0.8

chore(deps): bump micromatch from 4.0.5 to 4.0.8

gavindsouza committed:
Merge pull request #27276 from Mutantpenguin/patch-1

fix: :bug: show link titles for virtual link fields too

blaggacao committed:
docs: add extensive docstring to debug_on

blaggacao committed:
feat: Add --pdb and --pdb-on flags to FrappeTestCase

blaggacao committed:
feat: add list-sites command

blaggacao committed:
feat: add --pdb

TurkerTunali committed:
fix: Enhance the error message of the connected app authorization.

Error were "Invalid state". The reason was user who created the token. So we've added a little bit more context here.

(cherry picked from commit 4ad55b98c28d4c342808d166e7c124b89729607e)

blaggacao committed:
fix: test module + test case + test method name

blaggacao committed:
chore: lint

blaggacao committed:
Merge pull request #27650 from blaggacao/feat/list-sites-command

feat: add list-sites command

blaggacao committed:
Merge pull request #25734 from blaggacao/feat/debug-tests

feat: add debug tests

rmehta committed:
fix(ui): sortable sidebar, fixes to breadcrumbs etc

blaggacao committed:
fix: --pdb drop directly into test case frame

blaggacao committed:
Merge pull request #27651 from blaggacao/fix/stack-frame-drop

fix: --pdb drop directly into test case frame

akhilnarang committed:
Merge pull request #27646 from akhilnarang/bump-cryptography

chore(deps): bump cryptography

ruthra-kumar committed:
refactor: handle Issue and Lead separately on Communication

shariquerik committed:
Merge pull request #27656 from ruthra-kumar/handle_lead_and_issue_separately

refactor: handle Issue and Lead separately on Communication

akhilnarang committed:
Merge pull request #27608 from nikkothari22/increase-google-event-id-char-limit

fix: increase Google Calendar Event ID character limit

akhilnarang committed:
chore: update message

(cherry picked from commit 4e6581b0fc9006bdf3e869565ec59727556972c1)

gavindsouza committed:
Merge pull request #27618 from barredterra/refactor-data-field-validation

refactor: data field validation

frappe-pr-bot committed:
fix: sync translations from crowdin (#27638)

gavindsouza committed:
Merge branch 'develop' into set_link_titles

gavindsouza committed:
fix(search): Add label only if value exists

cogk committed:
fix(email): Fix unbound variable in make_error_msg

UnboundLocalError: local variable 'partial_mail' referenced before assignment

akhilnarang committed:
Merge pull request #27665 from frappe/mergify/bp/develop/pr-27649

fix: Enhance the error message of the connected app authorization. (backport #27649)

akhilnarang committed:
Merge pull request #27667 from cogk/fix-UnboundLocalError-partial_mail

fix(email): Fix unbound variable in make_error_msg

akhilnarang committed:
fix: don't let `getattr()` raise an `AttributeError`, provide a default

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27669 from akhilnarang/fix-27276

fix: don't let `getattr()` raise an `AttributeError`, provide a default

gavindsouza committed:
Merge branch 'develop' into set_link_titles

shariquerik committed:
fix: load apps_data in boot for desk

shariquerik committed:
Merge pull request #27673 from shariquerik/apps-page-1

fix: load apps_data in boot for setup wizard in desk

gavindsouza committed:
Merge pull request #27563 from gavindsouza/set_link_titles

fix(desk): Use doc.name as fallback title

akhilnarang committed:
fix: don't let getattr() raise an AttributeError, provide a default

Re-doing https://github.com/frappe/frappe/pull/27669 overwritten by https://github.com/frappe/frappe/pull/27563

gavindsouza committed:
Merge pull request #27676 from gavindsouza/develop

fix: don't let getattr() raise an AttributeError, provide a default

blaggacao committed:
feat: add value tracer for test debugging

blaggacao committed:
Merge pull request #27683 from blaggacao/feat/add-value-tracing-instrumentation

feat/add value tracing instrumentation

rmehta committed:
fix(ui): body sidebar is responsive

rmehta committed:
fix(minor): merge fixes

akhilnarang committed:
Merge pull request #26395 from barredterra/address-autocomplete

feat: address autocomplete dialog

akhilnarang committed:
Merge pull request #27051 from barredterra/user_doctype_permissions

feat: new report "User Doctype Permissions"

pa1narendra committed:
feat(auth): added keycloak as a social login provider

akhilnarang committed:
Revert "fix(ChildTable): apply depends_on on field update"

This reverts commit 895a81cccd3a855ec94b591f8b43ef6b8afb4845.

rmehta committed:
fix(minor): page can have different title

akhilnarang committed:
Merge pull request #27690 from akhilnarang/revert-grid-fix

fix: Revert "fix(ChildTable): apply depends_on on field update"

rmehta committed:
fix(test): workspace

rmehta committed:
fix(minor): workspace parent should be name, not title

rmehta committed:
fix(minor): workspace parent should be name, not title

rmehta committed:
Merge pull request #27621 from rmehta/app-switcher

feat: App switcher inside desk

blaggacao committed:
feat: Implement minutes_offset feature and add test cases

blaggacao committed:
fix: Add support for minutes-based notifications

frappe-pr-bot committed:
chore: update POT file (#27699)

git-avc committed:
refactor: use getattr fallback instead conditional to get translatable

Co-authored-by: Akhil Narang <me@akhilnarang.dev>

git-avc committed:
Merge branch 'frappe:develop' into fix-report-excel-translation

dependabot[bot] committed:
chore(deps): bump actions/download-artifact from 4.1.7 to 4.1.8

Bumps [actions/download-artifact](https://github.com/actions/download-artifact) from 4.1.7 to 4.1.8.
- [Release notes](https://github.com/actions/download-artifact/releases)
- [Commits](https://github.com/actions/download-artifact/compare/v4.1.7...v4.1.8)

---
updated-dependencies:
- dependency-name: actions/download-artifact
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

akhilnarang committed:
Merge pull request #27707 from frappe/dependabot/github_actions/actions/download-artifact-4.1.8

chore(deps): bump actions/download-artifact from 4.1.7 to 4.1.8

foolishdino committed:
fix(quick_list_widget):workflow-state-not-showing (#27596)

* fix(quick_list_widget):workflow-state-not-showing

when a document has workflow state, it should show workflow_status. Instead it doesn't.

existing code put a validation if status exist, but a document can have a workflow_state without a "status" field.

the correct approach is to separate the logic.

* fix(quick_list_widget):follow-suggestioin-to-remove-unnecessary-validation

Co-authored-by: Akhil Narang <me@akhilnarang.dev>

---------

Co-authored-by: Akhil Narang <me@akhilnarang.dev>

rmehta committed:
fix(style): print format

rmehta committed:
fix(style): print format

akhilnarang committed:
Merge pull request #27124 from git-avc/fix-report-excel-translation

fix: let's translate data on standard report excel/csv exporting

blaggacao committed:
fix: notification condition has no custom context (perf)

s-aga-r committed:
fix: prevent sending emails to System Manager when email notifications are disabled

blaggacao committed:
test: add more notification offset tests for edge cases

blaggacao committed:
Merge pull request #27722 from blaggacao/fix/notification-condition-context

fix: notification condition has no custom context (perf)

akhilnarang committed:
Merge pull request #27698 from blaggacao/chore/reuse-safe-exec-autocomplete

feat: subject processing to global server script config

rmehta committed:
fix(minor): default app title and logo

rmehta committed:
Merge pull request #27716 from rmehta/print-style-fixes

fix(style): print format

akhilnarang committed:
fix(setup_wizard): don't crash if a country doesn't have a timezone specified

Currently, this event is getting triggered by the country and its translated value
The translated value isn't present in the country info list, and so doesn't have any data

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27730 from akhilnarang/fix-country-selection-setup-wizard

fix(setup_wizard): don't crash if a country doesn't have a timezone specified

blaggacao committed:
chore: add no-stale label

blaggacao committed:
Merge pull request #27735 from blaggacao/chore/make-stalebot-more-bearable

chore: add no-stale label

blaggacao committed:
refactor!: make room in the frappe.config namespace

BRAKING: fappe.config namespace moved to
frappe.utils.modules namespace

blaggacao committed:
chore: simplify config calling

blaggacao committed:
Merge pull request #26737 from blaggacao/feat/cache-control

feat: add cache control

blaggacao committed:
Merge pull request #27740 from frappe/chore/simplify-config-calling

chore: simplify config calling

blaggacao committed:
chore: exempt draft prs from stale bot

blaggacao committed:
fix: assert equal on docs; compare dicts

blaggacao committed:
Merge pull request #26891 from blaggacao/feat/utm-analytics

feat(analytics): add basic utm nomenclature for analytics

blaggacao committed:
test: comment out due to uncontrollable side effects

blaggacao committed:
ci: split up install

when init fails, this separates long log sections and keeps github responsive

blaggacao committed:
ci: split up install prepare

blaggacao committed:
fix: semgrep complaints

blaggacao committed:
Merge pull request #27744 from frappe/chore/improve-ci-log-responsiveness-1

ci: split up install prepare

blaggacao committed:
Merge pull request #27743 from blaggacao/chore/improve-ci-log-responsiveness

ci: split up install

akhilnarang committed:
chore: reword description

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

blaggacao committed:
Merge pull request #27697 from blaggacao/feat/notification-with-minutes-offset

feat/notification with minutes offset

blaggacao committed:
Merge pull request #27742 from frappe/chore/no-stale-on-draft

chore: exempt draft prs from stale bot

akhilnarang committed:
fix(printview): ensure document title is a string

Nothing is stopping people from setting non-string types as their title

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27747 from akhilnarang/printview-doc-title-str

fix(printview): ensure document title is a string

Lyes7 committed:
fix: workflow typos

blaggacao committed:
Merge pull request #27738 from frappe/chore/make-config-namespace-available

refactor!: make room in the frappe.config namespace

blaggacao committed:
feat: friendly error wrapper on cli

blaggacao committed:
chore: also improve the pdb test mode

Ponnusamy1-V committed:
style: pre-commit

blaggacao committed:
fix!: execute command

BREAKING: return values of bench execute are not quoted anymore
For example, `bech execute frappe.bold --arg 'hi'`
- now returns: `<strong>hi</strong>`
- instead of: `"<strong>hi</strong>"`

blaggacao committed:
Merge pull request #27755 from Ponnusamy1-V/time_field_fix

fix: Time field not updating the manual edit to the time picker (editing the values directly in the input field)

blaggacao committed:
chore: fix all invocation of init for singledispatch -> min 1 positional arg

barredterra committed:
fix: form should pass df to formatter (#27756)

blaggacao committed:
fix: improve serializer

akhilnarang committed:
fix(rename_doc): check permissions for the specific document

A user can have access to write to a document without having access to the whole doctype

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27760 from akhilnarang/fix-validate-doc-rename-permission-check

fix(rename_doc): check permissions for the specific document

blaggacao committed:
Merge pull request #27759 from frappe/fix/json-serializer

fix: improve serializer

blaggacao committed:
Merge pull request #26753 from blaggacao/perf/more-critical-requests-cachable

perf: make more resources in the critical path cacheable

blaggacao committed:
refactor: type hint cli context

blaggacao committed:
Merge pull request #27766 from frappe/refactor/frappe-init-call-pre-pr

chore: fix all invocation of init for singledispatch -> min 1 positional arg

akhilnarang committed:
Merge pull request #27505 from dj12djdjs/fix-link-title-href

fix: show link title in generated form hrefs

akhilnarang committed:
chore: format

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27723 from s-aga-r/fix-system-manager-emails

fix: prevent sending emails to System Manager when email notifications are disabled

blaggacao committed:
fix: nicer error message on bench execute

blaggacao committed:
Merge pull request #27763 from frappe/fix/correct-execute-command

fix!: execute command

blaggacao committed:
Merge pull request #27753 from frappe/feat/add-friedly-error-handler-on-cli

feat: friendly error wrapper on cli

blaggacao committed:
Merge pull request #27765 from frappe/refactor/cli-context-lsp-support

refactor: type hint cli context

gavindsouza committed:
fix(delete_doc): Check if submittable before docstatus validation

blaggacao committed:
fix: align signature with backend; use count for perf on filters

blaggacao committed:
Merge pull request #27776 from frappe/perf/count-on-db-exists

fix: align signature with backend; use count for perf on filters

akhilnarang committed:
fix(setup_wizard): update wording for password field for non-admins

It's an optional field because the user already exists, clarify that
that this can just update their password, otherwise the original
password will still work.

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27782 from akhilnarang/update-password-message

fix(setup_wizard): update wording for password field for non-admins

akhilnarang committed:
chore(decrypt): mention the key in error if specificed

This would allow the user to know which key is failing, and just update that if required

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
chore(twofactor): specify key to display during decryption failures

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27786 from akhilnarang/improve-decryption-failed-message

chore(decrypt): mention the key in error if specificed

akhilnarang committed:
Merge pull request #27772 from gavindsouza/delete_doc-check_if_submittable

fix(delete_doc): Check if submittable before docstatus validation

akhilnarang committed:
Merge pull request #27746 from Lyes7/fix/fix-workflow-typos

fix: workflow typos

akhilnarang commented:
@shantanoo there's some conflicts now, can you resolve those, will merge it.

cogk committed:
fix(file): Remove illegal chars in file_name

- / and \ cannot work as file names
- % is not allowed verbatim in URLs
- ? and # are not allowed because confused with query/hash separators

akhilnarang committed:
Merge pull request #26693 from cogk/fix-attachment-name-with-percent

fix: Fix file/email attachment name containing percent sign

shantanoo committed:
Merge branch 'feature/sha-functions-for-server-scripts' of github.com:shantanoo/frappe into feature/sha-functions-for-server-scripts

shantanoo commented:
@akhilnarang something seems incorrect. Did 'git rebase develop' and there are 445 commits to be merged from this PR.

sumitbhanushali committed:
fix: dont fetch context for custom doctype based on custom module

sumitbhanushali committed:
Merge pull request #27795 from frappe/cust_mod_wf

fix: dont fetch context for custom doctype based on custom module

akhilnarang commented:
> @akhilnarang something seems incorrect. Did 'git rebase develop' and there are 445 commits to be merged from this PR.

@shantanoo Looks like you pulled your branch after the rebase instead of force pushing.

shantanoo commented:
Hmm, will delete the local branch and create branch of the same name again, and then force push the branch. Hope that will resolve the issue.

gavindsouza committed:
ci: Show which test record creation failed

ruthra-kumar committed:
fix: re-number idx on child table row removal

akhilnarang committed:
Merge pull request #27799 from ruthra-kumar/renumber_idx_on_remove

fix: re-number idx on child table row removal

blaggacao committed:
fix: migration to utm

blaggacao committed:
Merge pull request #27801 from frappe/fix/migration-patch-marketing-campaign

fix: migration to utm

akhilnarang committed:
chore: add back describe-database-table command

Partial revert of b169f8780ab543a21ee091c1d105c10d2c22f076

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

akhilnarang committed:
Merge pull request #27825 from akhilnarang/add-back-describe-cmd

chore: add back describe-database-table command

gavindsouza committed:
Merge pull request #27798 from gavindsouza/run-tests_verbosity

ci: Show which test record creation failed

barredterra committed:
feat: better filter labels (#27808)

frappe-pr-bot committed:
fix: sync translations from crowdin (#27679)

Mutantpenguin committed:
fix: :bug: escape special characters in entry_dn

entry_dn can contain special characters like parenthesis (e.g. to surround nicknames), but these will be interpreted as part of the filter syntax when not escaped and lead to an LDAPInvalidFilterError

akhilnarang committed:
refactor(keycloak): set API endpoint while getting list of oauth providers

urljoin() just adds the base URL to the relative path, here our base URL has an extra route in the path which isn't added in.

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

cogk committed:
fix(restore): Remove sandbox mode comment

https://github.com/MariaDB/server/commit/d20518168aff435a4843eebb108e5b9df24c19fb

cogk committed:
chore: Add type hints

akhilnarang committed:
Merge pull request #27689 from pa1narendra/social-login-key-keycloak

feat(auth): Added keycloak as a social login provider

akhilnarang committed:
Merge pull request #27831 from cogk/fix-db-restore-strip-mariadb-sandbox-mode-comment

fix(restore): Remove sandbox mode comment

venkat102 committed:
fix: show chart tooltip in report currency

venkat102 committed:
fix: update format

akhilnarang committed:
Merge pull request #27843 from aerele/report-chart-currency

fix: show chart tooltip in report currency

cogk committed:
fix(translate): Don't extract from gitignored files (#27498)

blaggacao committed:
fix: parent translation tests

blaggacao committed:
fix: Implement parent-child language precedence in get_all_translations

blaggacao committed:
Merge pull request #27852 from blaggacao/fix/parent-translation

fix: parent translation tests

barredterra committed:
feat!: enhance Language to become more of a Locale (#27178)

barredterra committed:
fix: money in words (#27190)

elshafei-developer committed:
fix: Replace "<br>" with "\n" (#27854)

akhilnarang committed:
Merge pull request #27829 from Mutantpenguin/patch-1

LDAP: escape special characters in `entry_dn` to prevent `LDAPInvalidFilterError`

shantanoo committed:
Merge branch 'feature/sha-functions-for-server-scripts' of github.com:shantanoo/frappe into feature/sha-functions-for-server-scripts
